qiskit/providers/ibmq/ibmqbackendservice.py

```diff
@@ -182,24 +183,29 @@ def jobs(
             else:
                 raise IBMQBackendValueError('unrecognized value for "status" keyword '
                                             'in job filter')
-            api_filter.update(this_filter)
+            filters.append(this_filter)
 
         if job_name:
-            api_filter['name'] = {"regexp": job_name}
+            filters.append({'name': {"regexp": job_name}})
 
-        # Create the date query according to the parameters specified.
-        if start_datetime or end_datetime:
-            date_filter = {'creationDate': {}}  # type: ignore[var-annotated]
-            if start_datetime:
-                date_filter['creationDate'].update({'gt': start_datetime.isoformat()})
-            if end_datetime:
-                date_filter['creationDate'].update({'lt': end_datetime.isoformat()})
-
-            api_filter.update(date_filter)
+        if start_datetime:
+            filters.append({'creationDate': {'gt': start_datetime.isoformat()}})
+        if end_datetime:
+            filters.append({'creationDate': {'lt': end_datetime.isoformat()}})
 
         if db_filter:
-            # status takes precedence over db_filter for same keys
-            api_filter = {**db_filter, **api_filter}
+            # Argument filters take precedence, remove duplicate `db_filter` keys.
+            for filter_ in filters:
+                for key in filter_:
+                    if key in db_filter:
+                        db_filter.pop(key, None)
+
+            # Add remaining `db_filter` items.
+            for key in db_filter:
+                filters.append({key: db_filter[key]})
```

Comment:
Instead of this complicated triple `for` loop, I think you can keep `api_filter` as a simple dictionary, and use `between` or `and` for the datetime range.

Reply:
The reason I decided to go with this (I do agree it is more complicated) was to ensure we kept using `lt` and `gt` for filtering when `start_datetime` and `end_datetime` are specified, since `between` uses `lte` and `gte`. But, I don't think it would be "too big a deal" to use `lte` or `gte` for filtering jobs and I'm sure the results returned would be the same, if not nearly identical. I'll go ahead and change it to use `between`, thanks @jyu00!